adapters/saml/core/src/main/java/org/keycloak/adapters/saml/RoleMappingsProviderUtils.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.adapters.saml;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.ServiceLoader;
+
+import org.jboss.logging.Logger;
+import org.keycloak.adapters.saml.config.SP;
+import org.keycloak.adapters.saml.config.parsers.ResourceLoader;
+
+/**
+ * Utility class that allows for the instantiation and configuration of role mappings providers.
+ *
+ * @author <a href="mailto:sguilhen@redhat.com">Stefan Guilhen</a>
+ */
+public class RoleMappingsProviderUtils {
+
+    private static final Logger logger = Logger.getLogger(RoleMappingsProviderUtils.class);
+
+    /**
+     * Loads the available implementations of {@code RoleMappingsProvider} and selects the provider that matches the id
+     * that was configured in {@code keycloak-saml.xml}. If no provider was configured for the SP then the default
+     * {@code PropertiesBasedRoleMapper} is instantiated. The selected provider is then initialized with the specified
+     * {@code SamlDeployment}, {@code ResourceLoader} and configuration as specified in {@code keycloak-saml.xml}.
+     *
+     * @param deployment a reference to the {@code SamlDeployment} that is being built.
+     * @param loader a reference to the {@code ResourceLoader} that allows the provider implementation to load additional
+     *               resources from the SP application WAR.
+     * @param providerConfig the provider configuration properties as configured in {@code keycloak-saml.xml}. Can contain
+ *                   an empty properties object if no configuration was specified for the provider.
+     * @return the configured
+     */
+    public static RoleMappingsProvider bootstrapRoleMappingsProvider(final SamlDeployment deployment, final ResourceLoader loader, final SP.RoleMappingsProviderConfig providerConfig) {
+        String providerId;
+        if (providerConfig == null || providerConfig.getId() == null) {
```

Comment:
We could. Reason why I'm always defaulting to the properties-based provider is that it works without any config (if you don't set it up and don't provide a properties file location it will simply return the same set of roles that is passed into it) and if one wants to add role mappings one must only drop a role-mappings.properties into WEB-INF and get it to work without further config.

Still we could chose to return null here - might be more natural than having a default that does nothing automatically installed. Will think about this and discuss with the rest of the team.

Reply:
I see your point. However, if you are adding a new role-mappings.properties file, it is not a big step to add also `<RoleMappingsProvider id="properties-based-role-mapper" />` to keycloak-saml.xml. To make things even more simple, it might be possible to make it work even like this: `<RoleMappingsProvider>` with id defaulting to `properties-based-role-mapper.`